portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -42,11 +44,14 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-
+    String maxCommentParam = request.getParameter("max-comments");
+    int maxNumComments = maxCommentParam == null ? Integer.MAX_VALUE : Integer.parseInt(maxCommentParam);
+    
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> commentsList = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    for (int i = 0; i < results.countEntities(FetchOptions.Builder.withDefaults()) && i < maxNumComments; i++) {
```

Comment:
Lets pull result.countEntities(..options..) into a local variable right before this loop 
Based on the javadocs for this api (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#countentities) --> it seems like it could be a somewhat expensive operation, so we shouldn't recompute at every loop iteration